sympy/matrices/common.py

```diff
@@ -2252,6 +2245,20 @@ def __mul__(self, other):
 
         matrix_multiply_elementwise
         """
+
+        return self.mul(other)
+
+    def mul(self, other, mulsimp=None):
```

Comment:
What about using the existing `Matrix.multiply` in https://docs.sympy.org/latest/modules/matrices/matrices.html#sympy.matrices.matrices.MatrixBase.multiply?

Reply:
The `mul()` function essentially took the body of `__mul__()` but allowed the extra `mulsimp` parameter so that is why I left it in `common.py`, I didn't want to transpose code from common into another module (like moving this code into `multiply`) and I also didn't want to add a `mulsimp` keyword to the `__mul__()` function since that would be non-standard.

Reply:
Maybe it would be better to have a separate function for the mulsimp multiply and mul can just dispatch to one or other function leaving the existing `_eval_matrix_mul` routines unmodified.

Reply:
The slowdown is due entirely to calling out to the `dotprodsimp()` function - even if I modify the function to do the simplest fastest dot product - it is still slow due to the overhead of calling the function and passing iterators I guess. So yes, what you say is the route I am taking now, only calling out to `dotprodsimp()` if needed. Otherwise the `_eval_matrix_mul` functions will essentially do what they did before.

Reply:
I actually meant by moving `MatrixBase.multiply` to `common.py`.
It should already have been, or we may end up with having a duplicate function

Reply:
Yeah, weird, `multiply_elementwise` is in `MatrixArithmetic` and `multiply` is in `MatrixBase` so they probably should both be in common. As for naming, as a function `multiply` only appears in `matrices.py` and `mathml.py` and does not seem to actually be called from anywhere in SymPy, whereas `mul` seems to be the common shorthand across many modules.

So how about I move the `multiply` into `common.py` alongside `multiply_elementwise` and leave the `mul` since it is such so common and we do wind up with a duplicate function but it doesn't actually harm anything.

Reply:
No, the duplicates will be regretted and may have to be deprecated in the future.
Even if you don't like the naming, it's better to respect the old one already came before.
Or you should better get agreed upon deprecating `Matrix.multiply` and introduce some shorthand.

Reply:
Ok, will go with `MatrixArithmetic.multiply` and nuke `mul` and `MatrixBase.multiply`.